File: iot_HTTP_server/models/action.js
'use strict'

const mongoose = require('mongoose')
const Schema = mongoose.Schema

const ActionSchema = Schema({
  deviceId: String,
  newState: String,
  room: String,
  deviceType: String,
  year: Number,
  month: Number,
  day: Number,
  hour: Number
})

module.exports = mongoose.model('Action', ActionSchema)

File: iOS_app/Home Automation App/Models/Action.swift
//
//  Action.swift
//  Home Automation App
//
//  Created by Xavier Bou on 4/7/20.
//  Copyright © 2020 Xavier Bou. All rights reserved.
//

import Foundation

class Action: Codable {
    let _id: String?
    let deviceId: String?
    let newState: String?
    let deviceType: String?
    let room: String?
    let year: Int?
    let month: Int?
    let day: Int?
    let hour: Int?
}

struct ActionsList: Codable {
    let action: [Action]?
}



